automates/model_assembly/interfaces.py

```diff
@@ -16,12 +16,24 @@ def extract_mentions(self, doc_path: str, out_path: str) -> dict:
         if not out_path.endswith(".json"):
             raise ValueError("/pdf_to_mentions requires an JSON output file")
 
-        res = requests.post(
-            f"{self.webservice}/pdf_to_mentions",
-            headers={"Content-type": "application/json"},
-            json={"pdf": doc_path, "outfile": out_path},
-        )
-        print(f"HTTP {res} for /pdf_to_mentions on {doc_path}")
+        if doc_path.endswith(".pdf"):
+            res = requests.post(
+                f"{self.webservice}/pdf_to_mentions",
+                headers={"Content-type": "application/json"},
+                json={"pdf": doc_path, "outfile": out_path},
+            )
+            print(f"HTTP {res} for /pdf_to_mentions on {doc_path}")
+
+        elif doc_path.endswith(".json"):
```

Comment:
@maxaalexeeva let's make this a little more specific.

I am creating all of the JSON files that include COSMOS processing output with the file-ending : `--COSMOS-data.json` (see examples of this in `ASKE-AutoMATES/Data/TR/COSMOS-pdf-output/`). Let's use that in the check here by doing the following:
```python
if ...:
  ...
elif doc_path.endswith("--COSMOS-data.json"):
    """< do COSMOS data processing here >"""
elif doc_path.endswith(".json"):
 """< do processing on non-COSMOS JSON here (such as the JSON saved results from parsing the ASCE paper with ScienceParse) >"""
else:
  raise ValueError(...)
```

This will ensure that we can still handle JSON output from ScienceParse while allowing us to distinguish that output from COSMOS processing output.

Reply:
@pauldhein, great call!